### PR TITLE
DM: card fixes and new implementations

### PR DESCRIPTION
**Pull request by dmikalova**
- Add FadingApparition, SlapdashEnamel, Starpatch card implementations
- Fix ArbiterVyynx, Gigantor2, HoardingTheGoods, Hydrogan2, KeyDrain, Klyyhnug, MonsterZero2, ProductiveTrash, TheGoldenQueen2, Zomok2
- Fix Cosmicrux (VM25)
- ReadyPhase and GameActions/RemoveTokenAction fixes
- CardImport and DeckService updates
- Client: CardImage and archonMaker fixes
- Add/update tests for new and fixed cards

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it adds a new token type and changes core game timing/logging around ready/reap/token-removal, which can affect many interactions; however changes are scoped and covered by new targeted tests.
> 
> **Overview**
> **Adds new gameplay pieces and card logic.** Implements `Fading Apparition`, `Slapdash Enamel`, and `Starpatch`, and fixes/adjusts several DM and VM25 cards (notably `Cosmicrux` now applies ready-trigger damage via an interrupt/change-event flow to ensure simultaneous resolution).
> 
> **Extends token and rules infrastructure.** Introduces the `corrosion` counter end-to-end (server actions for add/remove, client rendering/order, and a new `corrosion.svg` asset), refines `RemoveTokenAction` up-to prompting/log messaging, and updates `ReadyPhase` entrench ready prompting to respect `ready` restrictions.
> 
> **Data/import and tests.** Updates anomaly/special-card handling for DM set imports (`CardImport`, `DeckService`) and adds/extends unit tests for the new cards and the updated behavior (Cosmicrux timing, reap edge-cases, corrosion destruction, limited-amber capture selection, etc.).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b44a0cceb6f11a681c7d00a9817352feb5c6eb38. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: server/game/cards/11-VM25/Cosmicrux.js [hunk 1]
```diff
@@ -3,13 +3,20 @@ const Card = require('../../Card.js');
 class Cosmicrux extends Card {
     // After a player readies a creature, deal 1 to it.
     setupCardAbilities(ability) {
-        this.reaction({
+        this.interrupt({
```
Comment (dmikalova) [inline server/game/cards/11-VM25/Cosmicrux.js:6]: Not sure how, but this makes it so cosmicrux has all cards ready, and then deals damage to all of them simultaneously.

File: server/game/GameActions/RemoveTokenAction.js [hunk 2]
```diff
@@ -58,12 +58,22 @@ class RemoveTokenAction extends CardGameAction {
                         choices: Array.from(Array(event.amount + 1), (x, i) => i.toString()),
                         choiceHandler: (choice) => {
                             event.amount = parseInt(choice);
+                            let tokenLabel;
```
Comment (dmikalova) [inline server/game/GameActions/RemoveTokenAction.js:61]: This is for Transmutation messaging

File: server/game/gamesteps/ReadyPhase.js [hunk 3]
```diff
@@ -18,8 +18,15 @@ class ReadyPhase extends Phase {
     promptForEntrenched() {
         this.entrenchedToReady = new Set();
         const player = this.game.activePlayer;
+        const context = this.game.getFrameworkContext(player);
```
Comment (dmikalova) [inline server/game/gamesteps/ReadyPhase.js:21]: So cards that can't ready don't prompt entrench

File: server/game/cards/14-DM/HoardingTheGoods.js [hunk 4]
```diff
@@ -4,11 +4,33 @@ class HoardingTheGoods extends Card {
     // Play: Each friendly exhausted creature captures 1A.
     setupCardAbilities(ability) {
         this.play({
+            preferActionPromptMessage: true,
             gameAction: ability.actions.capture((context) => ({
                 amount: 1,
                 target: context.player.creaturesInPlay.filter((card) => card.exhausted)
             })),
-            effect: 'have each friendly exhausted creature capture 1 amber'
+            then: () => ({
```
Comment (strib) [inline server/game/cards/14-DM/HoardingTheGoods.js:12]: Hmm I would much rather just put all the creature names in one log message instead of having separate messages for each creature.  (Is that what this is doing?)

File: server/game/GameActions.js [hunk 5]
```diff
@@ -10,6 +10,8 @@ const Actions = {
     addDoomCounter: (propertyFactory) => new GameActions.AddTokenAction(propertyFactory, 'doom'),
     addAwakeningCounter: (propertyFactory) =>
         new GameActions.AddTokenAction(propertyFactory, 'awakening'),
+    addCorrosionCounter: (propertyFactory) =>
```
Comment (strib) [inline server/game/GameActions.js:13]: We'll need a corrosion icon too, right?

**Review by strib — APPROVED** [review]

File: server/game/cards/14-DM/HoardingTheGoods.js (hunk 4 above)
Comment (dmikalova) [reply server/game/cards/14-DM/HoardingTheGoods.js:12]: It just lists the creatures that captured

2026-05-05 10:28:26 player1 plays Hoarding the Goods
2026-05-05 10:28:26 player1 gains an amber due to Hoarding the Goods's bonus icon
2026-05-05 10:28:26 player1 plays Hoarding the Goods to have Caspart and Noxious Ionox capture 1 amber

**Review by strib — APPROVED** [review]
